tests/optimizer/test-function-eliminator-replace-array-value-with-hash-info.js

```diff
@@ -0,0 +1,32 @@
+var asm = (function(global, env, buffer) {
```

Comment:
it would be helpful to indent and whitespace the input test files the same as the output test files. That way comparing them to see what changes, when debugging, is easier. (Can either do it manually, or running `nodejs tools/js-optimizer` on each file should do the same, it will parse and then emit it the same as it does for the outputs.)